### ISSUE TITLE
Plugin calculates workload incorrectly by including all Jenkins labels

### ISSUE THREAD

**Issue by timvanderkooi** [jenkinsci/ec2-fleet-plugin#58]
Expectation:
- plugin calculates workload using only the labels that are assigned to the EC2 fleet configuration

Reality:
- plugin calculates workload using all available executors including nodes without EC2 Fleet labels

Example:
We noticed this because we have one local machine with 15 executors that are not always in use. Our build queue was building up to >10 builds and weren't triggering any new EC2 nodes. We waited nearly 30 mins - 1 hour for a new node until we decided to test marking the local machine temporarily offline. Moments later, new nodes were scheduled to handle the build queue.

**Comment by dhendo** [comment]
I'm seeing this too - we have a pool of local / non-spot fleet boxes in the jenkins instance that are not eligible to build a particular job.

Adding the ability to limit the nodes taken into consideration by label when calculating availability would help in certain circumstances:

- Multiple fleets, each pegged to a particular label (e.g. fleet 1 has label databasev1, fleet 2 databasev2, each fleet has an ami with the appropriate versions installed.). Branch A has databasev1 in the Jenkinsfile, so should cause fleet1 to scale up, even if there is capacity available in fleet 2.
- Local availability - we have nodes in-service for other unrelated purposes (e.g. building a separate product). These are guarded by a label e.g. productA. If a job for productB is queued, it currently looks like it won't scale out the fleet assigned to label productB, as there is capacity available (even if that capacity will not build the new job)

**Comment by dhendo** [comment]
@terma Is this likely the correct interpretation of what's likely to be happening, or is there some configuration I'm missing?

**Comment by terma** [comment]
hi, 

Ops, missed this issue, could you please specify what version of plugin do you use and Jenkins as well, btw do you have ```Restrict usage``` enabled?

**Comment by wosiu** [comment]
Do you know if the issue still exists? I'm considering using this plugin, but this one would be a strong blocker :(

**Comment by haugenj** [comment]
👋 @wosiu 

Using version 2.2.0 of the plugin I haven't had any problems with this. 

I just ran a quick test to verify too - 2 fleets with different labels. Fleet-1 has two instances available, Fleet-2 has 0 instances. I submitted a job with a label matching Fleet-2 and the plugin scaled up Fleet-2 to 1 instance on the next update. 

I think this issue is outdated, I'll close it out in a day or two if there's no more action here.
